Warnings with Xcode 8.2+

<!-- TEMPLATE FOR BUG REPORTS -->

### Expected behavior

Mantid builds without compiler errors or warnings

### Actual behavior

Warnings generated with Xcode 8.2+

```
/Users/svh/Documents/MantidProject/mantid/Framework/Kernel/src/EnabledWhenProperty.cpp:41:11: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
          std::move(std::make_shared<EnabledWhenProperty>(conditionOne)),
          ^
/Users/svh/Documents/MantidProject/mantid/Framework/Kernel/src/EnabledWhenProperty.cpp:41:11: note: remove std::move call here
          std::move(std::make_shared<EnabledWhenProperty>(conditionOne)),
          ^~~~~~~~~~                                                   ~
/Users/svh/Documents/MantidProject/mantid/Framework/Kernel/src/EnabledWhenProperty.cpp:42:11: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
          std::move(std::make_shared<EnabledWhenProperty>(conditionTwo)),
          ^
/Users/svh/Documents/MantidProject/mantid/Framework/Kernel/src/EnabledWhenProperty.cpp:42:11: note: remove std::move call here
          std::move(std::make_shared<EnabledWhenProperty>(conditionTwo)),
          ^~~~~~~~~~                                                   ~
2 warnings generated.
[119/127] Building CXX object Framework/Kernel/CMakeFiles/Kernel.dir/src/VisibleWhenProperty.cpp.o
/Users/svh/Documents/MantidProject/mantid/Framework/Kernel/src/VisibleWhenProperty.cpp:33:11: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
          std::move(std::make_shared<VisibleWhenProperty>(conditionOne)),
          ^
/Users/svh/Documents/MantidProject/mantid/Framework/Kernel/src/VisibleWhenProperty.cpp:33:11: note: remove std::move call here
          std::move(std::make_shared<VisibleWhenProperty>(conditionOne)),
          ^~~~~~~~~~                                                   ~
/Users/svh/Documents/MantidProject/mantid/Framework/Kernel/src/VisibleWhenProperty.cpp:34:11: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
          std::move(std::make_shared<VisibleWhenProperty>(conditionTwo)),
          ^
/Users/svh/Documents/MantidProject/mantid/Framework/Kernel/src/VisibleWhenProperty.cpp:34:11: note: remove std::move call here
          std::move(std::make_shared<VisibleWhenProperty>(conditionTwo)),
          ^~~~~~~~~~                                                   ~
2 warnings generated.
```

### Steps to reproduce the behavior

Build with Xcode 8.2+

### Platforms affected

MacOS 10.11+

These appear to have been introduced in #19264.

--- a/Framework/Kernel/src/EnabledWhenProperty.cpp
+++ b/Framework/Kernel/src/EnabledWhenProperty.cpp
@@ -37,10 +37,9 @@ EnabledWhenProperty::EnabledWhenProperty(
     : // This method allows the Python interface to easily construct these
       // objects
       // Copy the object then forward onto our move constructor
-      EnabledWhenProperty(
-          std::make_shared<EnabledWhenProperty>(conditionOne),
-          std::make_shared<EnabledWhenProperty>(conditionTwo),
-          logicOperator) {}
+      EnabledWhenProperty(std::make_shared<EnabledWhenProperty>(conditionOne),
+                          std::make_shared<EnabledWhenProperty>(conditionTwo),
+                          logicOperator) {}
 
 /** Multiple conditions constructor - takes two shared pointers to
 * EnabledWhenProperty objects and returns the product of them

--- a/Framework/Kernel/src/EnabledWhenProperty.cpp
+++ b/Framework/Kernel/src/EnabledWhenProperty.cpp
@@ -38,8 +38,8 @@ EnabledWhenProperty::EnabledWhenProperty(
       // objects
       // Copy the object then forward onto our move constructor
       EnabledWhenProperty(
-          std::move(std::make_shared<EnabledWhenProperty>(conditionOne)),
-          std::move(std::make_shared<EnabledWhenProperty>(conditionTwo)),
+          std::make_shared<EnabledWhenProperty>(conditionOne),
+          std::make_shared<EnabledWhenProperty>(conditionTwo),
           logicOperator) {}
 
 /** Multiple conditions constructor - takes two shared pointers to

--- a/Framework/Kernel/src/VisibleWhenProperty.cpp
+++ b/Framework/Kernel/src/VisibleWhenProperty.cpp
@@ -30,8 +30,8 @@ VisibleWhenProperty::VisibleWhenProperty(
     : // For the python interface to be able to easily copy objects in
       // Make a deep copy and turn into a shared pointer and forward on
       VisibleWhenProperty(
-          std::move(std::make_shared<VisibleWhenProperty>(conditionOne)),
-          std::move(std::make_shared<VisibleWhenProperty>(conditionTwo)),
+          std::make_shared<VisibleWhenProperty>(conditionOne),
+          std::make_shared<VisibleWhenProperty>(conditionTwo),
           logicOperator) {}
 
 /** Multiple conditions constructor - takes two shared pointers to

--- a/Framework/Kernel/src/VisibleWhenProperty.cpp
+++ b/Framework/Kernel/src/VisibleWhenProperty.cpp
@@ -29,10 +29,9 @@ VisibleWhenProperty::VisibleWhenProperty(
     const VisibleWhenProperty &conditionTwo, eLogicOperator logicOperator)
     : // For the python interface to be able to easily copy objects in
       // Make a deep copy and turn into a shared pointer and forward on
-      VisibleWhenProperty(
-          std::make_shared<VisibleWhenProperty>(conditionOne),
-          std::make_shared<VisibleWhenProperty>(conditionTwo),
-          logicOperator) {}
+      VisibleWhenProperty(std::make_shared<VisibleWhenProperty>(conditionOne),
+                          std::make_shared<VisibleWhenProperty>(conditionTwo),
+                          logicOperator) {}
 
 /** Multiple conditions constructor - takes two shared pointers to
 * VisibleWhenProperty objects and returns the product of them

